### DISCUSSION TITLE
Rename equations to formulas everywhere

### DISCUSSION THREAD

**Discussion by jakubkaczor** [typst/typst#3681] [General]
### Description

The name for the equation element and the associated functions is misleading. Not every formula that can be typeset in the equation element is an equation, nor equivalence. Merriam-Webster's dictionary, at the time of writing, gives the following definitions for formula and equation.

- formula [1]

> a general fact, rule, or principle expressed in usually mathematical symbols

> a group of symbols (such as letters and numbers) associated to express facts or data (such as the number and kinds of teeth in the jaw) concisely

- equation [2]

> a usually formal statement of the [equality](https://www.merriam-webster.com/dictionary/equality) or equivalence of mathematical or logical expressions

---
I wasn't sure if this issue should be classified as a bug (in design?), feature request, or a discussion.

[1]: https://www.merriam-webster.com/dictionary/formula
[2]: https://www.merriam-webster.com/dictionary/equation

### Use Case

It would improve accuracy of representing equation, as currently named, element's meaning in the name.

**Comment by laurmaedje** [comment]
FWIW, it used to be called `formula` and I renamed it to `equation` because I had found various sources saying that a formula is a special type of equation with just one quantity on the left. I did a quick Google search now and I still found some of those, while others said that formula is the more general term.

Ultimately, it seems to me that there is no generally agreed upon general term for a bunch of math.
Since LaTeX also uses the term equation, I picked that one. 

One idea we had though is to rename `equation` to just `math`, replacing the `math` module, and all the other math stuff ends up on the associated scope.

**Reply by emilyyyylime** [reply]
I am for renaming equations. Formula has some potential to be considered more general than equation, but leaning into the single unifying property that is math-mode parsing sounds like the best option as far as I'm concerned.

**Reply by jakubkaczor** [reply]
I have not studied logic extensively, but I have seen definitions for formulas and well-formed formulas. One of them is contained in Peter B. Andrews' “An Introduction to Mathematical Logic and Type Theory: To Truth Through Proof” (https://link.springer.com/book/10.1007/978-94-015-9934-4). I think this is an example of formalization of what we usually mean by a formula. I do suspect that the definition given by Andrews may not be universal, because it uses set theory. Furthermore, I expect formulas to be defined before set theory, but I also suspect that other definitions aren't much different in the meaning, and the meaning is far from equating.

To me, math appears to be worse for the following reasons. I would expect math to be much more inconsistently defined. Also, not every formula, defined in the manner similar to the Andrews', is for mathematics, and I believe you should be able to typeset such in Typst.

**Comment by Enivex** [comment]
I'm against this. Equation is the established terminology in LaTeX, which is the tool that is universally used today. While this doesn't mean that we should follow LaTeX in everything, this smells like change for the sake of change.

That being said. _If_ the name were to change I'd prefer math over formula.

**Reply by jakubkaczor** [reply]
> Equation is the established terminology in LaTeX, which is the tool that is universally used today.

I am not sure if I understand your argument. Do you suggest that equation shouldn't be renamed, because it would be inconsistent with LaTeX, and in this case, being consistent with LaTeX is important, because it is universally used today? If so, why in this case you consider this significant?

**Reply by Enivex** [reply]
> > Equation is the established terminology in LaTeX, which is the tool that is universally used today.
> 
> I am not sure if I understand your argument. Do you suggest that equation shouldn't be renamed, because it would be inconsistent with LaTeX, and in this case, being consistent with LaTeX is important, because it is universally used today? If so, why in this case you consider this significant?

I'm saying that typst is framed as a LaTeX replacement, and equation is such a ubiquitous name that there should be a good reason to change it. I'm not seeing one.

**Comment by glyn** [comment]
I'm in favour of being able to replace "Equation" if only for internationalisation. E.g. in a Spanish paper, I'd want to see "Ecuación 3" rather than "Equation 3".

Another use case for replacing "Equation" is when an equation is really a definition.

**Reply by laurmaedje** [reply]
When writing Spanish, you should set `text.lang` to `"es"` like this:
```typ
#set text(lang: "es")
#set math.equation(numbering: "(1)")

$ x + y y = z $ <my>

See @my for details.
```

This should change to "Ecuación" automatically. For full control, you can use the `supplement` parameter:

```typ
#set math.equation(numbering: "(1)", supplement: [My Custom Ecuación])
```

This discussion here was about the scripting name for `math.equation`, not the output.

**Reply by jakubkaczor** [reply]
Actually, the discussion is about renaming everywhere, including output. So that instead of “Equation” rendering in text, there would be “Formula” by default.

**Reply by glyn** [reply]
Well I certainly think the output needs more fine-grained control. (I'm not using Spanish, I was simply trying to motivate better control, but that can probably be motivated by English-only use cases, such as "definitions" which look like equations.)

**Reply by laurmaedje** [reply]
You can always override the supplement if need be. With `#set math.equation(supplement: none)` for example, you'll not get the "Equation" or alternatively with `@my[Definition]` you'll get a definition specifically at the place where you write the reference. Or if you want only specific equations to be definitions, you can apply the set rule scoped locally.
